src/webServiceClient.ts

```diff
@@ -74,97 +77,94 @@ export default class WebServiceClient {
     const parsedPath = `/minfraud/v2.0/${path}`;
     const url = `https://${this.host}${parsedPath}`;
 
-    const options = {
-      auth: `${this.accountID}:${this.licenseKey}`,
+    const controller = new AbortController();
+    const timeoutId = setTimeout(() => controller.abort(), this.timeout);
+
+    const options: RequestInit = {
+      body: postData,
       headers: {
         Accept: 'application/json',
-        'Content-Length': Buffer.byteLength(postData),
+        Authorization:
+          'Basic ' +
+          Buffer.from(`${this.accountID}:${this.licenseKey}`).toString(
+            'base64'
+          ),
+        'Content-Length': Buffer.byteLength(postData).toString(),
         'Content-Type': 'application/json',
         'User-Agent': `minfraud-api-node/${version}`,
       },
-      host: this.host,
       method: 'POST',
-      path: parsedPath,
-      timeout: this.timeout,
+      signal: controller.signal,
     };
 
-    return new Promise((resolve, reject) => {
-      const req = https.request(options, (response) => {
-        let data = '';
-
-        response.on('data', (chunk) => {
-          data += chunk;
-        });
-
-        response.on('end', () => {
-          if (response.statusCode && response.statusCode === 204) {
-            return resolve();
-          }
-
-          try {
-            data = JSON.parse(data);
-          } catch {
-            return reject(this.handleError({}, response, url));
-          }
-
-          if (response.statusCode && response.statusCode !== 200) {
-            return reject(
-              this.handleError(data as ResponseError, response, url)
-            );
-          }
-
-          return resolve(new modelClass(data));
-        });
-      });
-
-      req.on('error', (err: NodeJS.ErrnoException) => {
-        return reject({
-          code: err.code,
-          error: err.message,
-          url,
-        } as WebServiceClientError);
-      });
-
-      req.write(postData);
-
-      req.end();
-    });
+    let data;
+    try {
+      const response = await fetch(url, options);
+
+      if (!response.ok) {
+        return Promise.reject(await this.handleError(response, url));
+      }
+
+      if (response.status === 204) {
+        return Promise.resolve();
```

Comment:
Is this different from just a bare return?

```suggestion
        return;
```